### PR TITLE
TG-921 TG-925 Refactoring sh scripts

### PR DESCRIPTION
**Pull request by filippo-20tab**

### CODE REVIEW

**Comment by filippo-20tab** [comment]
By using the `docker:20` image instead of the previous `docker/compose:debian-1.29.2` image, `wget` is now available (neither `curl` nor `wget` are available in the other image). `sed` was used to extract the JSON field from the secret to avoid installing `jq` (not available in either image).

**Comment by trottomv** [comment]
> In the `docker/compose:debian-1.29.2` image , neither `curl` nor `wget` are available. Therefore, it was replaced with `docker:20`, which comprises all the `docker compose` functionalities, plus `wget` is available in it. As `jq` was unavailable in either image, in order not to install the extra package, a workaround with `sed` and a regex was used to extract the JSON field from the secret.

Pay attention to the `docker compose` functionality plugin, because It's not available in the custom ci cd runner with shell executor

File: {{cookiecutter.project_dirname}}/scripts/ci_sentry.sh [hunk 1]
```diff
@@ -2,26 +2,27 @@
...
 
-  export VAULT_TOKEN=`curl --silent --request POST --data "role=${VAULT_ROLE}" --data "jwt=${CI_JOB_JWT_V2}" ${VAULT_ADDR%/}/v1/auth/gitlab-jwt/login | jq -r .auth.client_token`
+  vault_token=$(curl --silent --request POST --data "role=${VAULT_ROLE}" --data "jwt=${CI_JOB_JWT_V2}" "${VAULT_ADDR%/}"/v1/auth/gitlab-jwt/login | jq -r .auth.client_token)
 
-  export SENTRY_AUTH_TOKEN=`curl --silent --header "X-Vault-Token: ${VAULT_TOKEN}" ${VAULT_ADDR%/}/v1/${PROJECT_SLUG}/envs/${ENV_NAME}/sentry | jq -r .data.sentry_auth_token`
-
-  export SENTRY_DSN=`curl --silent --header "X-Vault-Token: ${VAULT_TOKEN}" ${VAULT_ADDR%/}/v1/${PROJECT_SLUG}/envs/${ENV_NAME}/${SERVICE_SLUG}/sentry | jq -r .data.sentry_dsn`
+  SENTRY_AUTH_TOKEN=$(curl --silent --header "X-Vault-Token: ${vault_token}" "${VAULT_ADDR%/}"/v1/"${PROJECT_SLUG}"/envs/"${ENV_NAME}"/sentry | jq -r .data.sentry_auth_token)
+  SENTRY_DSN=$(curl --silent --header "X-Vault-Token: ${vault_token}" "${VAULT_ADDR%/}"/v1/"${PROJECT_SLUG}"/envs/"${ENV_NAME}"/"${SERVICE_SLUG}"/sentry | jq -r .data.sentry_dsn)
+  export SENTRY_AUTH_TOKEN
```
Comment (trottomv) [inline {{cookiecutter.project_dirname}}/scripts/ci_sentry.sh:14]: consider to add an `unset` for sensitive environment variables at point where they become no longer necessary

File: {{cookiecutter.project_dirname}}/scripts/deploy/vault.sh [hunk 2]
```diff
@@ -1,14 +1,18 @@
...
+    secrets_data=$(echo "${secrets_data}" | jq --argjson new_data "${secret_data}" '. * $new_data')
 done
 
-if [ "${TERRAFORM_BACKEND}" == "terraform-cloud" ]; then
-    export TFC_TOKEN=`curl --silent --header "X-Vault-Token: ${VAULT_TOKEN}" ${VAULT_ADDR%/}/v1/${PROJECT_SLUG}-tfc/creds/default | jq -r .data.token `
+echo "${secrets_data}" > "${TERRAFORM_VARS_DIR%/}"/vault-secrets.tfvars.json
+
+if [ "${TERRAFORM_BACKEND}" = "terraform-cloud" ]; then
+    TFC_TOKEN=$(curl --silent --header "X-Vault-Token: ${vault_token}" "${VAULT_ADDR%/}"/v1/"${PROJECT_SLUG}"-tfc/creds/default | jq -r .data.token)
+    export TFC_TOKEN
```
Comment (trottomv) [inline {{cookiecutter.project_dirname}}/scripts/deploy/vault.sh:17]: consider to add an `unset` for sensitive environment variables at point where they become no longer necessary

File: {{cookiecutter.project_dirname}}/scripts/ci_pact.sh [hunk 3]
```diff
@@ -3,13 +3,17 @@
...
 
-  export PACT_BROKER_BASE_URL=`echo ${pact_secrets} | jq -r .pact_broker_base_url`
-  export PACT_BROKER_PASSWORD=`echo ${pact_secrets} | jq -r .pact_broker_password`
-  export PACT_BROKER_USERNAME=`echo ${pact_secrets} | jq -r .pact_broker_username`
+  PACT_BROKER_BASE_URL=$(echo "${pact_secrets}" | jq -r .pact_broker_base_url)
+  PACT_BROKER_PASSWORD=$(echo "${pact_secrets}" | jq -r .pact_broker_password)
+  PACT_BROKER_USERNAME=$(echo "${pact_secrets}" | jq -r .pact_broker_username)
+
+  export PACT_BROKER_BASE_URL
+  export PACT_BROKER_PASSWORD
```
Comment (trottomv) [inline {{cookiecutter.project_dirname}}/scripts/ci_pact.sh:15]: consider to add an `unset` for sensitive environment variables at point where they become no longer necessary

File: {{cookiecutter.project_dirname}}/scripts/ci_sentry.sh (hunk 1 above)
Comment (filippo-20tab) [reply {{cookiecutter.project_dirname}}/scripts/ci_sentry.sh:14]: GitLab runner deals with this already, there are a lot of sensitive env vars involved in a single run of a CI job (e.g. the JWT token itself), and they are all handled correctly by the runner. I would not worry about this or anything of the sort.

File: {{cookiecutter.project_dirname}}/scripts/deploy/vault.sh (hunk 2 above)
Comment (filippo-20tab) [reply {{cookiecutter.project_dirname}}/scripts/deploy/vault.sh:17]: GitLab runner deals with this already, there are a lot of sensitive env vars involved in a single run of a CI job (e.g. the JWT token itself), and they are all handled correctly by the runner. I would not worry about this or anything of the sort.

File: {{cookiecutter.project_dirname}}/scripts/ci_pact.sh (hunk 3 above)
Comment (filippo-20tab) [reply {{cookiecutter.project_dirname}}/scripts/ci_pact.sh:15]: GitLab runner deals with this already, there are a lot of sensitive env vars involved in a single run of a CI job (e.g. the JWT token itself), and they are all handled correctly by the runner. I would not worry about this or anything of the sort.

**Comment by filippo-20tab** [comment]
> > In the `docker/compose:debian-1.29.2` image , neither `curl` nor `wget` are available. Therefore, it was replaced with `docker:20`, which comprises all the `docker compose` functionalities, plus `wget` is available in it. As `jq` was unavailable in either image, in order not to install the extra package, a workaround with `sed` and a regex was used to extract the JSON field from the secret.
> 
> Pay attention to the `docker compose` functionality plugin, because It's not available in the custom ci cd runner with shell executor

The proposed code has been extensively tested (both with GitLab runner shell and docker executors) and works fine. There is no problem whatsoever with using the `docker-compose` command in the `docker:20` image.

**Review by daniele20tab — APPROVED** [review]

**Pull request merged**